harness/determined/profiler.py

```diff
@@ -505,20 +510,25 @@ def measure(self, batch_idx):
             handle = pynvml.nvmlDeviceGetHandleByIndex(i)
             try:
                 util = pynvml.nvmlDeviceGetUtilizationRates(handle)
-                gpu_util = util.gpu
-
-            except pynvml.NVMLError as err:
-                # TODO: Log this error
-                continue
-            measurements[handle] = Measurement(timestamp, batch_idx, gpu_util)
+                measurements[handle] = Measurement(timestamp, batch_idx, util.gpu)
+            except pynvml.NVMLError as e:
+                logging.warning(f"failed to sample GPU utilization: {e}")
         return measurements
 
 
-# TODO: Haven't figured out how to collect GPU memory usage yet
-class GpuMemory:
-    pass
-
+class GpuMemoryCollector():
+    def __init__(self):
+        pynvml.nvmlInit()
+        self.num_gpus = pynvml.nvmlDeviceGetCount()
```

Comment:
i'll fix this up once i do the bumpenv for the environments and actually have nvml